lxd/storage/drivers/driver_btrfs.go

```diff
@@ -0,0 +1,1514 @@
+		// reverse operation. So instead we shamelessly mount using the
+		// block device path at the time of pool creation.
+		err = tryMount(source, GetPoolMountPath(d.name), "btrfs", mountFlags, mountOptions)
+	} else {
+		_, err = d.Mount()
+	}
+	if err != nil {
+		return err
+	}
+
+	// Create default subvolumes.
+	subvolumes := []string{
+		filepath.Join(poolMntPoint, "containers"),
+		filepath.Join(poolMntPoint, "containers-snapshots"),
+		filepath.Join(poolMntPoint, "custom"),
+		filepath.Join(poolMntPoint, "custom-snapshots"),
+		filepath.Join(poolMntPoint, "images"),
+		filepath.Join(poolMntPoint, "virtual-machines"),
+		filepath.Join(poolMntPoint, "virtual-machines-snapshots"),
+	}
+
+	for _, subvol := range subvolumes {
+		err := btrfsSubvolumeCreate(subvol)
+		if err != nil {
+			return fmt.Errorf("Could not create btrfs subvolume: %s", subvol)
+		}
+	}
+
+	return nil
+}
+
+// Delete removes the storage pool from the storage device.
+func (d *btrfs) Delete(op *operations.Operation) error {
+	source := d.config["source"]
+
+	if source == "" {
+		return fmt.Errorf("no \"source\" property found for the storage pool")
+	}
+
+	if strings.HasPrefix(source, "/") {
+		source = shared.HostPath(d.config["source"])
+	}
+
+	poolMntPoint := GetPoolMountPath(d.name)
+
+	// Delete default subvolumes.
+	subvolumes := []string{
+		filepath.Join(poolMntPoint, "containers"),
+		filepath.Join(poolMntPoint, "containers-snapshots"),
+		filepath.Join(poolMntPoint, "custom"),
+		filepath.Join(poolMntPoint, "custom-snapshots"),
+		filepath.Join(poolMntPoint, "images"),
+		filepath.Join(poolMntPoint, "virtual-machines"),
+		filepath.Join(poolMntPoint, "virtual-machines-snapshots"),
+	}
+
+	for _, subvol := range subvolumes {
+		err := btrfsSubvolumesDelete(subvol)
+		if err != nil {
+			return fmt.Errorf("Could not delete btrfs subvolume: %s", subvol)
+		}
+	}
+
+	_, err := d.Unmount()
+	if err != nil {
+		return err
+	}
+
+	if filepath.IsAbs(source) {
+		var err error
+		cleanSource := filepath.Clean(source)
+		sourcePath := shared.VarPath("disks", d.name)
+		loopFilePath := sourcePath + ".img"
+
+		if cleanSource == loopFilePath {
+			// This is a loop file so simply remove it.
+			err = os.Remove(source)
+		} else {
+			if !isBtrfsFilesystem(source) && isBtrfsSubvolume(source) {
+				err = btrfsSubvolumesDelete(source)
+			}
+		}
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	return wipeDirectory(poolMntPoint)
+}
+
+func (d *btrfs) Validate(config map[string]string) error {
+	return nil
+}
+
+func (d *btrfs) Update(changedConfig map[string]string) error {
+	return nil
+}
+
+// Mount mounts the storage pool.
+func (d *btrfs) Mount() (bool, error) {
+	source := d.config["source"]
+
+	if source == "" {
+		return false, fmt.Errorf("no \"source\" property found for the storage pool")
+	}
+
+	if strings.HasPrefix(source, "/") {
+		source = shared.HostPath(d.config["source"])
+	}
+
+	poolMntPoint := GetPoolMountPath(d.name)
```

Comment:
Call the variable `path` for consistency with other drivers.

Trying to keep the ordering and general style similar so it's easier to follow and apply fixes to all of them, we don't want them to feel as isolated as they currently are.